### PR TITLE
Add shared channel v1.0 APIs

### PR DESCRIPTION
**Pull request by devjha-ms**
Expose shared channel v1.0 APIs

### CODE REVIEW

File: api-reference/v1.0/api/associatedteaminfo-list.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+<!-- {
+  "blockType": "ignored"
+}
+-->
+``` http
+GET /users/{user-id}/teamwork/associatedTeams
+```
+
+## Optional query parameters
+This method does not currently support the [OData query parameters](/graph/query-parameters) to customize the response.
```
Comment (shantanu1singh) [inline api-reference/v1.0/api/associatedteaminfo-list.md:40]: Maybe rephrase to 'This method currently does not support....' ?

File: api-reference/v1.0/api/sharedwithchannelteaminfo-get.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+author: "devjha-ms"
+doc_type: "apiPageType"
+ms.localizationpriority: high
+ms.prod: "microsoft-teams"
+---
+
+# Get sharedWithChannelTeamInfo
+Namespace: microsoft.graph
+
+Get a [team](../resources/sharedwithchannelteaminfo.md) which has been shared a specified [channel](../resources/channel.md). This operation is allowed only for channels with a **membershipType** value of `shared`.
```
Comment (shantanu1singh) [inline api-reference/v1.0/api/sharedwithchannelteaminfo-get.md:13]: which has been shared with a specified... ?

File: api-reference/v1.0/api/sharedwithchannelteaminfo-list-allowedmembers.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+---
+
+# List allowedMembers
+Namespace: microsoft.graph
+
+Get the list of [conversationMembers](../resources/conversationmember.md) who can access a shared [channel](../resources/channel.md).
+
+This method does not return the following [conversationMembers](../resources/conversationmember.md) from the [team](../resources/team.md):
+- Users with `Guest` role
+- Users who are externally authenticated in the tenant
```
Comment (shantanu1singh) [inline api-reference/v1.0/api/sharedwithchannelteaminfo-list-allowedmembers.md:17]: Should we be linking to some other documentation here on what externally authenticated means?

File: api-reference/v1.0/api/sharedwithchannelteaminfo-list.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+author: "devjha-ms"
+doc_type: "apiPageType"
+ms.localizationpriority: high
+ms.prod: "microsoft-teams"
+---
+
+# List sharedWithChannelTeamInfo
+Namespace: microsoft.graph
+
+Get the list of [teams](../resources/sharedwithchannelteaminfo.md) that has been shared a specified [channel](../resources/channel.md). This operation is allowed only for channels with a **membershipType** value of `shared`.
```
Comment (shantanu1singh) [inline api-reference/v1.0/api/sharedwithchannelteaminfo-list.md:13]: '...that has been shared with a specified..' ?

File: api-reference/v1.0/api/user-list-joinedteams.md [hunk 5]
```diff
@@ -14,6 +14,7 @@ Namespace: microsoft.graph
 
 
 Get the [teams](../resources/team.md) in Microsoft Teams that the user is a direct member of.
+**Note**: This API doesn't return the host team of the shared channel that the user is a direct member of.
```
Comment (shantanu1singh) [inline api-reference/v1.0/api/user-list-joinedteams.md:17]: Might be worth linking to the associatedteaminfo-list.md file here?

Saying something like:
'To retrieve the host teams of shared channels that the user is a direct member of, use the list associatedTeamInfo API'

File: api-reference/v1.0/resources/channel.md [hunk 6]
```diff
@@ -43,6 +45,12 @@ where files are shared, and where tabs are added.
...
 |[Provision channel email address](../api/channel-provisionemail.md) |[provisionChannelEmailResult](../resources/provisionchannelemailresult.md)| Provision an email address for the channel.|
 |[Remove channel email address](../api/channel-removeemail.md) | None | Remove the email address of the channel.|
+|[Remove incoming channel](../api/team-delete-incomingchannels.md) | None| Remove an incoming channel.|
+|[List teams shared with channel](../api/sharedwithchannelteaminfo-list.md)|[sharedWithChannelTeamInfo](../resources/sharedwithchannelteaminfo.md) collection|Get the list of teams shared with the channel.|
+|[Get team shared with channel](../api/sharedwithchannelteaminfo-get.md)|[sharedWithChannelTeamInfo](../resources/sharedwithchannelteaminfo.md)|Get a team that is shared with the channel.|
+|[Unshare channel with team](../api/sharedwithchannelteaminfo-delete.md)|None|Unshare a channel with a team.|
+|[List allowed members](../api/sharedwithchannelteaminfo-list-allowedmembers.md)|[conversationMember](../resources/conversationmember.md) collection|Get the list of team members who have access to the shared channel.|
```
Comment (shantanu1singh) [inline api-reference/v1.0/resources/channel.md:52]: should this be 'List allowed members in shared channel'? Since it's scoped to shared channels only?

File: api-reference/v1.0/resources/channel.md [hunk 7]
```diff
@@ -43,6 +45,12 @@ where files are shared, and where tabs are added.
...
+|[Check user access](../api/channel-doesuserhaveaccess.md)|Boolean|Check whether a user has access to a shared channel or not.|
```
Comment (shantanu1singh) [inline api-reference/v1.0/resources/channel.md:53]: should this be 'Check user access to shared channel'? Since it's scoped to shared channels only?

File: api-reference/v1.0/resources/channel.md [hunk 8]
```diff
@@ -43,6 +45,12 @@ where files are shared, and where tabs are added.
 |[Remove tab from channel](../api/channel-delete-tabs.md) | None | Remove (unpin) a tab from a channel.|
 |[Provision channel email address](../api/channel-provisionemail.md) |[provisionChannelEmailResult](../resources/provisionchannelemailresult.md)| Provision an email address for the channel.|
 |[Remove channel email address](../api/channel-removeemail.md) | None | Remove the email address of the channel.|
+|[Remove incoming channel](../api/team-delete-incomingchannels.md) | None| Remove an incoming channel.|
```
Comment (shantanu1singh) [inline api-reference/v1.0/resources/channel.md:48]: Maybe reword to 'Remove a channel shared with a team'. It would align well with what we have documented above for List incoming channels too

File: api-reference/v1.0/resources/sharedwithchannelteaminfo.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+ms.localizationpriority: high
+ms.prod: "microsoft-teams"
+doc_type: resourcePageType
+---
+
+# sharedWithChannelTeamInfo resource type
+
+Namespace: microsoft.graph
+
+Represents a information for a [team](team.md) with which a channel is shared. A [team](team.md) can be shared multiple channels.
```
Comment (shantanu1singh) [inline api-reference/v1.0/resources/sharedwithchannelteaminfo.md:14]: Remove 'a' -> Represents information for a team..

Comment (shantanu1singh) [reply api-reference/v1.0/resources/sharedwithchannelteaminfo.md:14]: Also, rephrase thee second sentence - 'A team can be shared with multiple channels

File: api-reference/v1.0/resources/team.md [hunk 10]
```diff
@@ -39,7 +39,12 @@ For more information about working with groups and members in teams, see [Use th
...
 |[Complete migration](../api/team-completemigration.md)|[team](team.md)| Removes migration mode from the team and makes the team available to users to post and read messages.|
+|[List all channels](../api/team-list-allchannels.md)|[channel](../resources/channel.md) collection|Get the list of [channels](../resources/channel.md) either in this **team** or shared with this **team** (incoming channels).|
+|[List channels](../api/channel-list.md)|[channel](../resources/channel.md) collection|Get the list of [channels](../resources/channel.md) in a **team**.|
+|[List incoming channels](../api/team-list-incomingchannels.md)|[channel](../resources/channel.md) collection|Get the list of [channels](../resources/channel.md) shared with this **team**.|
+|[Remove incoming channel](../api/team-delete-incomingchannels.md) | None| Remove an incoming channel.|
 |[List apps installed in team](../api/team-list-installedapps.md) | [teamsAppInstallation](teamsappinstallation.md) collection | List apps installed in a team.|
```
Comment (shantanu1singh) [inline api-reference/v1.0/resources/team.md:48]: Same comment here regarding the description. Consider changing it to - 'Remove an incoming channel (channel shared with the team)'

**Review by shantanu1singh — APPROVED** [review]

**Comment by JarbasHorst** [comment]
Hello @devjha-ms, I reviewed your PR and left some comments. Please, take a look when you will have time.

Also, to avoid merge conflicts, please merge PR [17453](https://github.com/microsoftgraph/microsoft-graph-docs/pull/17453) into your branch to include my fixes and before addressing my comments.

File: api-reference/v1.0/api/channel-doesuserhaveaccess.md [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+
+The following is an example of a request.
+
+<!-- {
+  "blockType": "request",
+  "name": "channel_doesuserhaveaccess"
+}
+-->
+``` http
+GET https://graph.microsoft.com/v1.0/teams/{0fddfdc5-f319-491f-a514-be1bc1bf9ddc}/channels/19:33b76eea88574bd1969dca37e2b7a819@thread.skype/doesUserHaveAccess(userId='6285581f-484b-4845-9e01-60667f8b12ae')
```
Comment (JarbasHorst) [inline api-reference/v1.0/api/channel-doesuserhaveaccess.md:73]: For clarity, the first GUID shouldn't be bounded by curly brackets, right? => {0fddfdc5-f319-491f-a514-be1bc1bf9ddc} 

Please, fix this issue and apply the fix to all examples in this API topic.

**Comment by JarbasHorst** [comment]
Hello, I continued with the review work for your PR and left some additional comments. Please, take a look when you will have time.

Also, to avoid merge conflicts, please merge PR [17458](https://github.com/microsoftgraph/microsoft-graph-docs/pull/17458) into your branch to include my fixes and before addressing my new comments.

File: api-reference/v1.0/api/associatedteaminfo-list.md (hunk 1 above)
Comment (devjha-ms) [reply api-reference/v1.0/api/associatedteaminfo-list.md:40]: This change is no required for consistency across graph documentation.

File: api-reference/v1.0/api/sharedwithchannelteaminfo-list-allowedmembers.md (hunk 3 above)
Comment (devjha-ms) [reply api-reference/v1.0/api/sharedwithchannelteaminfo-list-allowedmembers.md:17]: I could not find any documentation explaining "externally authenticated". This has a literal meaning that the user is not authenticated in current tenant.

File: api-reference/v1.0/resources/channel.md (hunk 6 above)
Comment (devjha-ms) [reply api-reference/v1.0/resources/channel.md]: We don't specify the channel type in API title.

**Comment by devjha-ms** [comment]
Hi @JarbasHorst @shantanu1singh @Lauragra,

Fixed review comments. Please review the latest changes. 

Thanks

File: api-reference/v1.0/api/channel-post.md [hunk 12]
```diff
@@ -386,6 +386,53 @@ Content-type: application/json
 }
 ```
 
+### Example 5: Create a shared channel on behalf of a user
```
Comment (JarbasHorst) [inline api-reference/v1.0/api/channel-post.md:389]: In the corresponding [channel-post beta topic](https://docs.microsoft.com/en-us/graph/api/channel-post), we have the following note. Should the same note also be added to the channel-post v1.0 topic?

image

Comment (devjha-ms) [reply api-reference/v1.0/api/channel-post.md:389]: Thanks for noticing this. I have removed the note from beta API as this doc PR will be merged as part of shared channel GA.

**Comment by JarbasHorst** [comment]
@Lauragra, this PR is ready for final review.

**Review by Lauragra — APPROVED** [review]

**Comment by devjha-ms** [comment]
Hi @Lauragra,
I have merged a commit to resolve conflicts. 
Please review the latest changes and merge the PR and the feature is already available now.

Regards,
Devi Shankar
